cluster-install/install.sh

```diff
@@ -1183,6 +1218,53 @@ EOF
     echo "$yaml_labels"
 }
 
+# k8s_get_resource will get the given k8s resource in the given format and echo the output.
+# If an error occurs, a problem will be added if the error is NOT "NotFound". This is to prevent
+# misdiagnosing all errors as "the resource wasn't found", which can be dangerous (especially
+# considering we look for the existence of the TORC to determine whether Trident is installed or not).
+k8s_get_resource() {
+    local -r resource="$1"
+    local -r namespace="${2:-""}"
+    local -r format="${3:-"json"}"
+
+    [ -z "$resource" ] && fatal "no resource given"
+
+    local -a args=()
+    [ -n "$namespace" ] && args+=("-n" "$namespace")
+
+    local -r output="$(kubectl get "$resource" "${args[@]}" -o "$format" 2> "$__ERR_FILE")"
+    local -r captured_err="$(get_captured_err)"
+    if [ -n "$output" ] && [ -z "$captured_err" ]; then
+        echo "$output"
+        return 0
+    fi
+
+    local -r base_msg="A failure occurred when checking if resource '$resource' (namespace: $namespace) exists"
+    if echo "$captured_err" | grep -q "NotFound" &> /dev/null; then
+        logdebug "got NotFound error for resource '$resource', letting it through" >& 2
+    elif echo "$captured_err" | grep -q "error executing jsonpath"; then
+        logdebug "got jsonpath error for resource '$resource', letting it through" >& 2
+    elif [ -n "$captured_err" ]; then
+        logdebug "got unrecognized error for resource '$resource': $captured_err" >& 2
+        add_problem "$base_msg: $(echo "$captured_err" | tail -n 1)"
+    else
+        add_problem "$base_msg: unknown error"
+    fi
```

Comment:
This is where we handle the different errors. Both `NotFound` and `jsonpath` errors will return an exit code of 1 (indicating the resource wasn't found), while others add an actual problem, which will cause the script to exit and display the error returned by k8s.

For example:
```
* A failure occurred when checking if resource 'crd/volumesnapshotclasses.snapshot.storage.k8s.io' exists: Unable to connect to the server: dial tcp: some.bad.api.server.url: no such host
```